File: src/main/java/org/r1/gde/service/ComputingResult.java
package org.r1.gde.service;

import java.util.ArrayList;
import java.util.List;

import lombok.Data;

@Data
public class ComputingResult {

	boolean error = false;
	String errorMsg = "";

	boolean bytesXlsInProgress = false;
	boolean bytesDbfInProgress = false;
	boolean xlsComputationOk = false;
	boolean perfDbfComputationOk = false;
	boolean debitDbfComputationOk = false;

	boolean decSent = false;
	boolean bvDecSent = false;
	boolean bvExuSent = false;
	boolean exuSent = false;

	int objRetComputeProgress = 0;
	int retComputeProgress = 0;
	int cassisComputeProgress = 0;
	int q100ComputeProgress = 0;

	boolean objRetComputeOk = false;
	boolean retComputeOk = false;
	boolean cassisComputeOk = false;
	boolean q100ComputeOk = false;

	int xlsComputeProgress = 0;
	int perfDbfComputeProgress = 0;
	int debitDbfComputeProgress = 0;

	// en import
	List<String> inDbfBVDecWarns = new ArrayList<String>();
	List<String> inDbfDecWarns = new ArrayList<String>();
	List<String> inDbfBVExuWarns = new ArrayList<String>();
	List<String> inDbfExuWarns = new ArrayList<String>();
	// génération excel
	List<String> paramWarns = new ArrayList<String>();
	List<String> objRetWarns = new ArrayList<String>();
	List<String> retBassinsWarns = new ArrayList<String>();
	List<String> q100Warns = new ArrayList<String>();
	List<String> cassisWarns = new ArrayList<String>();
	// génération DBF
	List<String> bvDecDBFWarns = new ArrayList<String>();
	List<String> exuDBFWarns = new ArrayList<String>();

}

File: src/main/frontend/src/app/ComputingResult.ts
export class ComputingResult {

	error: boolean = false;
	errorMsg: string = "";

	bytesXlsInProgress: boolean = false;
	bytesDbfInProgress: boolean = false;
	xlsComputationOk: boolean = false;
	perfDbfComputationOk: boolean = false;
	debitDbfComputationOk: boolean = false;

	objRetComputeProgress: number = 0;
	retComputeProgress: number = 0;
	cassisComputeProgress: number = 0;
	q100ComputeProgress: number = 0;

	objRetComputeOk: boolean = false;
	retComputeOk: boolean = false;
	cassisComputeOk: boolean = false;
	q100ComputeOk: boolean = false;

	xlsComputeProgress: number = 0;
	perfDbfComputeProgress: number = 0;
	debitDbfComputeProgress: number = 0;

	inDbfBVDecWarns: string[];
	inDbfDecWarns: string[];
	inDbfBVExuWarns: string[];
	inDbfExuWarns: string[];
	paramWarns: string[];
	objRetWarns: string[];
	retBassinsWarns: string[];
	q100Warns: string[];
	cassisWarns: string[];
	bvDecDBFWarns: string[];
	exuDBFWarns: string[];
	
}
